examples/multipage/src/main.rs

```diff
@@ -0,0 +1,101 @@
+#[macro_use]
+extern crate lazy_static;
+
+use std::convert::Infallible;
+use std::net::SocketAddr;
+use hyper::{Body, Method, Request, Response, Server, StatusCode};
+use hyper::service::{make_service_fn, service_fn};
+use multipage::Route;
+
+lazy_static! {
+    pub static ref TEMPLATES: tera::Tera = {
+        let mut tera = match tera::Tera::new("examples/multipage/src/templates/*.html") {
+            Ok(t) => t,
+            Err(e) => {
+                println!("Parsing error(s): {}", e);
+                ::std::process::exit(1);
+            }
+        };
+        // disable autoescaping completely
+        tera.autoescape_on(vec![]);
+        tera
+    };
+}
```

Comment:
Wow! What is this? Can you explain what's going on here?

Reply:
I added a comment on the `TEMPLATES` value to explain how it is used. If you're referring to the `lazy_static` macro it is https://github.com/rust-lang-nursery/lazy-static.rs which

> Using this macro, it is possible to have statics that require code to be executed at runtime in order to be initialized. This includes anything requiring heap allocations, like vectors or hash maps, as well as anything that requires non-const function calls to be computed.

The short version is that this is setting up a way to dynamically output HTML contents at server runtime without having to define the full HTML in a string of the server code. It is probably overkill for this example but I was exploring how I could use this rather than writing specifically for the example.